### PR TITLE
[py2py3] Migration at level scr/python/A/B/C - slice 12

### LINKED ISSUE #10136: [py2py3] Migration at level scr/python/A/B/C - slice 12
#### Intro

Our purpose is to continue with the gradual migration. 
This is the 12-th slice in which the migration has been divided into. 
We have split migration units at level `src/python/A/B/C`.

A detailed plan of the migration at this depth is available 
at the dmwm mkdocs [detailed plan page](https://cms-dmwm-test.docs.cern.ch/py2py3/plan_detailed/), 
with summary information on top.

#### Impact of the new feature

We start migrating the 12-th bulk of migration units 
at src/python/A/B/C level.

```plaintext
                           name  nfiles  gradual  future_pm
     WMCore_WorkQueue_WorkQueue       1     True         20
WMCore_MicroService_DataStructs       4    False         76
```

#### Is your feature request related to a problem? Please describe.

The end goal is to have WMCore py2 and py3 compatible.

#### Describe the solution you'd like

Run futurize on the first bulk of units 
and then apply the necessary manual changes.

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #10136 

#### Status

ready

#### Related PRs

* Previous migration PR: #10299  (slice11, issue #10135 )
* Next migration PR: #10310  (slice13, issue #10137  )

#### Description

Run futurize and some manual changes on the first batch of src/python/A/B/C.

Other changes that require some comments:

- `src/python/WMCore/WorkQueue/WorkQueue.py`: here `str()` is used extensively in py2 to simply format log messages, so I would like to use the "native string" approach and not touch it. However, we also need to compare a variable with both unicode and bytes strings types. This is why I decided to decouple the import `from builtins import str as newstr, bytes` to use `(newstr, bytes)` instead of `basestring`, but without affecting all the extensive use of `str()` that likely will not require any change when switching to py3

#### Is it backward compatible (if not, which system it affects?)

It should be. However,
- `src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py` checks if the python2 types `str` or `unicode` are in a list. If the list is not built properly, this is likely to fail. There is no workaround, we can only modernize this code and fix accordingly where the list `data['type']` is built from. This may require some testing that goes beyond unit tests.
  - tracked in #10323 

#### External dependencies / deployment changes

Requires python-future in both py2 and py3 environments.

### CODE REVIEW

File: src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py [hunk 1]
```diff
@@ -120,8 +122,8 @@ def _wfParse(self):
                     self.extDoc[keyName] = {}
                     for item in data['values']:
                         self.extDoc[keyName].update(item)
-                elif (isinstance(data['type'], tuple) and (str in data['type'] or unicode in data['type'])) or \
-                     (data['type'] is str or data['type'] is unicode):
+                elif (isinstance(data['type'], tuple) and (bytes in data['type'] or str in data['type'])) or \
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py:125]: rhetorical comment, what a horrible line!

File: src/python/WMCore/WorkQueue/WorkQueue.py [hunk 2]
```diff
@@ -12,6 +12,9 @@
 
 from __future__ import division, print_function
 
+from builtins import str as str_, bytes
```
Comment (amaltaro) [inline src/python/WMCore/WorkQueue/WorkQueue.py:15]: I've seen you using `from builtins import str as newstr` in slice 11. Should we adopt the same pattern in here as well?

**Review by amaltaro — CHANGES_REQUESTED** [review]
Dario, I think there is one minor change that we can make to keep things more consistent. If you make that change, feel free to squash it with the first commit.

File: src/python/WMCore/WorkQueue/WorkQueue.py (hunk 2 above)
Comment (mapellidario) [reply src/python/WMCore/WorkQueue/WorkQueue.py:15]: Uh, you are correct, I definitely lacked consistency here! fixed!

File: src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py (hunk 1 above)
Comment (mapellidario) [reply src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py:125]: I also lit up a candle when I pushed this, praying it does not break. Oh, let me create an issue to track this, we should definitely explicitely check that this work in py3.

Comment (mapellidario) [reply src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py:125]: Created #10323 to track this

File: src/python/WMCore/WorkQueue/WorkQueue.py (hunk 2 above)
Comment (mapellidario) [reply src/python/WMCore/WorkQueue/WorkQueue.py]: Pushed new commit with this change

**Review by amaltaro — APPROVED** [review]
Thanks for the prompt fix, Dario. Please squash those commits and it's good to go.

File: src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py (hunk 1 above)
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py:125]: A rhetorical reply. Thank you @amaltaro

**Pull request merged**